polaris/polaris/sep10auth/views.py

```diff
@@ -29,150 +35,179 @@
+
+        If the source account does not exist, verify that the keypair used as the
+        source for the challenge transaction has signed the challenge. This is
+        sufficient because newly created accounts have their own keypair as signer
+        with a weight greater than the default thresholds.
+        """
+        server_key = settings.SIGNING_KEY
+        net = settings.STELLAR_NETWORK_PASSPHRASE
+
+        logger.info("Validating challenge transaction")
+        try:
+            tx_envelope, account_id = read_challenge_transaction(
+                envelope_xdr, server_key, net
+            )
+        except InvalidSep10ChallengeError as e:
+            err_msg = f"Error while validating challenge: {str(e)}"
+            logger.error(err_msg)
+            raise ValueError(err_msg)
+
+        try:
+            account = settings.HORIZON_SERVER.load_account(account_id)
+        except NotFoundError:
+            logger.warning(
+                "Account does not exist, using client's master key to verify"
+            )
+            try:
+                verify_challenge_transaction_signed_by_client(
+                    envelope_xdr, server_key, net
+                )
+            except InvalidSep10ChallengeError as e:
+                logger.info(
+                    f"Missing or invalid signature(s) for {account_id}: {str(e)})"
+                )
+                raise ValueError(str(e))
+            else:
+                logger.info("Challenge verified using client's master key")
+                return
+
+        signers = account.load_ed25519_public_key_signers()
+        threshold = account.thresholds.med_threshold
+        try:
+            signers_found = verify_challenge_transaction_threshold(
+                envelope_xdr, server_key, net, threshold, signers
+            )
+        except InvalidSep10ChallengeError as e:
+            logger.info(str(e))
+            raise ValueError(str(e))
+
+        logger.info(f"Challenge verified using account signers: {signers_found}")
+
+    @staticmethod
+    def _generate_jwt(request: Request, envelope_xdr: str) -> str:
+        """
+        Generates the JSON web token from the challenge transaction XDR.
+
+        See: https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0010.md#token
+        """
+        issued_at = time.time()
+        transaction_envelope, source_account = read_challenge_transaction(
+            envelope_xdr, settings.SIGNING_KEY, settings.STELLAR_NETWORK_PASSPHRASE
+        )
+        logger.info(
+            f"Challenge verified, generating SEP-10 token for account {source_account}"
+        )
+        hash_hex = binascii.hexlify(transaction_envelope.hash()).decode()
+        jwt_dict = {
+            "iss": request.build_absolute_uri("/auth"),
+            "sub": source_account,
+            "iat": issued_at,
+            "exp": issued_at + 24 * 60 * 60,
+            "jti": hash_hex,
```

Comment:
The `jti` is no longer required in SEP-10. It was removed in v1.2.0. stellar/stellar-protocol@353329faf6ace26b4db6d58c16bed58f23c34dc8

Reply:
I'm going to resolve this comment. Because Polaris is already in use it might not be safe to remove the `jti` and it's not really relevant to the change you're trying to make in this PR. It doesn't hurt to keep it.